src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js

```diff
@@ -282,70 +253,81 @@ class PopoverReportActionContextMenu extends React.Component {
+        setIsDeleteCommentConfirmModalVisible(true);
+    };
+
+    useImperativeHandle(ref, () => ({
+        showContextMenu,
+        hideContextMenu,
+        showDeleteModal,
+        hideDeleteModal,
+        isActiveReportAction,
+        instanceID,
+        runAndResetOnPopoverHide,
+    }));
+
+    const reportAction = reportActionRef.current;
+
+    return (
+        <>
+            <PopoverWithMeasuredContent
+                isVisible={isPopoverVisible}
+                onClose={hideContextMenu}
+                onModalShow={runAndResetOnPopoverShow}
+                onModalHide={runAndResetOnPopoverHide}
+                anchorPosition={popoverAnchorPosition.current}
+                animationIn="fadeIn"
+                disableAnimation={false}
+                animationOutTiming={1}
+                shouldSetModalVisibility={false}
+                fullscreen
+                withoutOverlay
+                anchorRef={anchorRef}
+            >
+                <BaseReportActionContextMenu
+                    isVisible
+                    type={typeRef.current}
+                    reportID={reportIDRef.current}
+                    reportActionID={reportActionIDRef.current}
+                    draftMessage={reportActionDraftMessageRef.current}
+                    selection={selectionRef.current}
+                    isArchivedRoom={isRoomArchived}
+                    isChronosReport={isChronosReportEnabled}
+                    isPinnedChat={isChatPinned}
+                    isUnreadChat={hasUnreadMessages}
+                    anchor={contextMenuTargetNode}
+                    contentRef={contentRef}
+                    originalReportID={originalReportIDRef.current}
                 />
-            </>
-        );
-    }
+            </PopoverWithMeasuredContent>
+            <ConfirmModal
+                title={translate('reportActionContextMenu.deleteAction', {reportAction})}
+                isVisible={isDeleteCommentConfirmModalVisible}
+                shouldSetModalVisibility={shouldSetModalVisibilityForDeleteConfirmation}
+                onConfirm={confirmDeleteAndHideModal}
+                onCancel={hideDeleteModal}
+                onModalHide={() => {
+                    reportIDRef.current = '0';
+                    reportActionRef.current = {};
+                    callbackWhenDeleteModalHide.current();
+                }}
+                prompt={translate('reportActionContextMenu.deleteConfirmation', {action: reportAction})}
+                confirmText={translate('common.delete')}
+                cancelText={translate('common.cancel')}
+                danger
+            />
+        </>
+    );
 }
 
-PopoverReportActionContextMenu.propTypes = propTypes;
+PopoverReportActionContextMenu.displayName = 'PopoverReportActionContextMenu';
 
-export default withLocalize(PopoverReportActionContextMenu);
+export default forwardRef(PopoverReportActionContextMenu);
```

Comment:
Sorry for the late review. Just one comment

I prefer forwardRef to be combined with component declarations

```js
const PopoverReportActionContextMenu = forwareRef(function PopoverReportActionContextMenu(props, ref) {})
```

If you use a named function `function PopoverReportActionContextMenu` inside `forwardRef`, I think you don't have to do `PopoverReportActionContextMenu.displayName = 'PopoverReportActionContextMenu';`

Reply:
According to the code style guide we should do it [this way](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#forwarding-refs), so I'm leaving it as is.